From 897919e3f797d92da6f4e1874f1a1b51f96b83ab Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:35 -0400
Subject: [powerpc] Make slb_shadow a local

Message-id: <1410545655-205645-227-git-send-email-dzickus@redhat.com>
Patchwork-id: 94203
O-Subject: [RHEL7 PATCH 226/626] powerpc: Make slb_shadow a local
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 1a8f6f97ea4dbaaa21b05cae2dacea47e4aea37b
Author: Jeremy Kerr <jk@ozlabs.org>
Date:   Thu Dec 5 11:31:08 2013 +0800

    powerpc: Make slb_shadow a local

    The only external user of slb_shadow is the pseries lpar code, and it
    can access through the paca array instead.

    Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/lppaca.h b/arch/powerpc/include/asm/lppaca.h
index 844c28d..d0a2a2f 100644
--- a/arch/powerpc/include/asm/lppaca.h
+++ b/arch/powerpc/include/asm/lppaca.h
@@ -132,8 +132,6 @@ struct slb_shadow {
  } save_area[SLB_NUM_BOLTED];
 } ____cacheline_aligned;
 
-extern struct slb_shadow slb_shadow[];
-
 /*
  * Layout of entries in the hypervisor's dispatch trace log buffer.
  */
diff --git a/arch/powerpc/kernel/paca.c b/arch/powerpc/kernel/paca.c
index 0620eaa..9095a6f 100644
--- a/arch/powerpc/kernel/paca.c
+++ b/arch/powerpc/kernel/paca.c
@@ -99,7 +99,7 @@ static inline void free_lppacas(void) { }
  * 3 persistent SLBs are registered here.  The buffer will be zero
  * initially, hence will all be invaild until we actually write them.
  */
-struct slb_shadow slb_shadow[] __cacheline_aligned = {
+static struct slb_shadow slb_shadow[] __cacheline_aligned = {
  [0 ... (NR_CPUS-1)] = {
   .persistent = cpu_to_be32(SLB_NUM_BOLTED),
   .buffer_length = cpu_to_be32(sizeof(struct slb_shadow)),
diff --git a/arch/powerpc/platforms/pseries/lpar.c b/arch/powerpc/platforms/pseries/lpar.c
index c8fbef2..49e4e0c 100644
--- a/arch/powerpc/platforms/pseries/lpar.c
+++ b/arch/powerpc/platforms/pseries/lpar.c
@@ -92,7 +92,7 @@ void vpa_init(int cpu)
   * PAPR says this feature is SLB-Buffer but firmware never
   * reports that.  All SPLPAR support SLB shadow buffer.
   */
- addr = __pa(&slb_shadow[cpu]);
+ addr = __pa(paca[cpu].slb_shadow_ptr);
  if (firmware_has_feature(FW_FEATURE_SPLPAR)) {
   ret = register_slb_shadow(hwcpu, addr);
   if (ret)
-- 
1.7.1